src/Host.cpp

```diff
@@ -802,15 +802,14 @@ bool Host::installPackage(const QString& fileName, int module )
                 pLabel->setText( tr( "Unpacking package:\n\"%1\"\nplease wait..." ).arg( packageName ) );
             }
         }
+        pUnzipDialog->hide(); // Must hide to change WindowModality
         pUnzipDialog->setWindowTitle( tr( "Unpacking" ) );
-        pUnzipDialog->hide();
         pUnzipDialog->setWindowModality( Qt::ApplicationModal );
         pUnzipDialog->show();
         qApp->processEvents();
         pUnzipDialog->raise();
-        pUnzipDialog->show(); // Must do this to ensure modality is applied
-        pUnzipDialog->update();
-        qApp->processEvents(); // Try to ensure we are on top of any other dialogs
+        pUnzipDialog->repaint(); // Force a redraw
```

Comment:
Not really, though Ubuntu uses compiz. I've tried it on two different setups - one with proprietary nVidia and another with Intel, and the issue is present in both. Windows doesn't have the label issue though.

I'll test it on OSX and if it's not an issue there, we can park this minor UI issue.